assets/src/vo/date-time/duration.js

```diff
@@ -0,0 +1,567 @@
+				keys( omit( values, unitNames ) ).join()
+			);
+			this[ privateProperties.isValid ] = false;
+		}
+		return valuesToSet;
+	}
+
+	/**
+	 * Used to set the internal "private" values property.
+	 *
+	 * @param {Object} values
+	 * @access private
+	 */
+	[ privateMethods.setValues ]( values ) {
+		this[ privateProperties.durationValues ] = {};
+		unitNames.forEach( ( unit ) => {
+			this[ privateProperties.durationValues ][ unit ] = values[ unit ] ||
+				0;
+		} );
+	}
+
+	/**
+	 * Used to set the values "private" property from a moment.Duration object.
+	 *
+	 * @param {moment.Duration} duration
+	 * @access private
+	 */
+	[ privateMethods.populateValuesFromDuration ]( duration ) {
+		const setValues = {};
+		unitNames.forEach( ( unit ) => {
+			setValues[ unit ] = duration[ unit ]();
+		} );
+		this[ privateMethods.setValues ]( setValues );
+	}
+
+	/**
+	 * Returns an array of accessor names (that in turn are used for generating
+	 * private properties).
+	 *
+	 * @access private
+	 * @return {string[]}  Array of accessor names.
+	 */
+	[ privateMethods.getAllUnitNames ]() {
+		return [
+			...unitNames,
+			...derivativeUnitNames,
+		];
+	}
+
+	/**
+	 * Creates getters for the Duration instance from the accessor names.
+	 * @access private
+	 */
+	[ privateMethods.createGetters ]() {
+		this[ privateMethods.getAllUnitNames ]().forEach(
+			( accessorName ) => {
+				// creates accessor for getting the value via a property
+				// eg. instance.hours
+				Object.defineProperty( this, accessorName, {
+					get() {
+						if ( derivativeUnitNames.indexOf( accessorName ) > -1 ) {
+							return this[ privateProperties.duration ][ accessorName ]();
+						}
+						return this
+							[ privateProperties.durationValues ]
+							[ accessorName ] ||
+							0;
+					},
+				} );
+				// creates `as*` methods.
+				// eg `instance.asHours` would return the given duration
+				// expressed as the hours unit.
+				// note for units such as "years" and "months", this uses what
+				// is termed as "longterm" calculation. Longterm is based on
+				// a 400 year cycle averaging out the days in a month and
+				// days in a year over that cycle.
+				// @link https://github.com/moment/moment/blob/develop/src/lib/duration/bubble.js#L52
+				const asMethodName = 'as' + capitalize( accessorName );
+				Object.defineProperty( this, asMethodName, {
+					get() {
+						return () => {
+							return this[ privateProperties.duration ]
+								[ asMethodName ]();
+						};
+					},
+				} );
+			}
+		);
+	}
+
+	/**
+	 * Exposes the value of locale.
+	 * eg. instance.locale
+	 * @return {string} The locale string.
+	 */
+	get locale() {
+		return this[ privateProperties.duration ].locale();
+	}
+
+	/**
+	 * Indicates whether the current Duration instance represents a valid
+	 * duration.
+	 *
+	 * @return {boolean} True means the Duration instance is valid.
+	 */
+	get isValid() {
+		return this[ privateProperties.isValid ];
```

Comment:
See my reply to your earlier comment.  Instanceof check won't work here because it doesn't matter what you feed `moment.duration` it will always get constructed.

Reply:
ya after reading that comment it appears to me that we don't need the `privateProperties.isValid` property at all and this could simply be:

```js
get isValid() {
	return this[ privateProperties.duration ].isValid();
}
```

since that seems to achieve the exact same thing (ie: returns false if object was creating via simple object using bad parameters).

Reply:
It is not the same.   I don't think you are understanding things correctly.  Did you try any of the examples I gave in my earlier comment?